src/client/network/mqtt/mqtt_mosquitto.c

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <mosquitto.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "client/network/mqtt/mqtt.h"
+
+struct mqtt_client {
+  struct mosquitto *mosq;
+  mqtt_client_config_t *config;
+  void (*mqtt_callback)(mqtt_client_event_t *event, void *userdata);
+  void *cb_userdata;
+};
+
+/* Callback called when the client receives a CONNACK message from the broker. */
+void on_mqtt_connect(struct mosquitto *mosq, void *client, int reason_code) {
+  int rc;
+  /* Print out the connection result.*/
+  printf("[%s:%d]: Connect Mqtt: %s\n", __func__, __LINE__, mosquitto_connack_string(reason_code));
+  // Create an event with eventid corresponding to Mqtt Connected
+  mqtt_client_event_t *mqtt_event = (mqtt_client_event_t *)malloc(sizeof(mqtt_client_event_t));
+  if (reason_code == 0) {
+    mqtt_event->event_id = MQTT_CONNECTED;
+  } else {
+    mqtt_event->event_id = MQTT_ERROR;
+    char *con_ack = mosquitto_connack_string(reason_code);
```

Comment:
please fix ‘const’ qualifier warnings, it means `data` element is the `char const *` in mqtt_client_event_t

Reply:
Done.